packages/medusa/src/services/draft-order.ts

```diff
@@ -340,6 +353,19 @@ class DraftOrderService extends TransactionBaseService {
           }
         }
 
+        const generateItemsCartData = {
+          ...createdCart,
+          items: generatedLineItems,
+        } as unknown as Cart
+
+        await Promise.all(
+          generatedLineItems.map(async (item) => {
+            return await this.lineItemAdjustmentService_
+              .withTransaction(transactionManager)
+              .createAdjustmentForLineItem(generateItemsCartData, item)
```

Comment:
question: Does this conflict with the adjustments creation being performed in `applyDiscount` on line 278?

Reply:
Not it does not because at that moment there is no items in the rawCart since they have been destructured before. One thing I am asking myself, would it be better to call applyDiscount on the localCartData instead. So that we benefit from all the validation etc. I ll have a look at it and in that case i ll remove the l278 loop in favor of moving it down here wdyt?

Reply:
Yeah, that was also what I had in mind. Might make sense to use that directly to go through all the additional logic we have around creating adjustments :)

Reply:
I ll do the changes first thing tomorrow 💪